webapp/channels/src/components/threading/channel_threads/thread_footer/thread_footer.tsx

```diff
@@ -45,7 +45,7 @@ function ThreadFooter({
     const thread = useSelector((state: GlobalState) => getThreadOrSynthetic(state, post));
 
     useEffect(() => {
-        if (threadIsSynthetic(thread) && thread.is_following && thread.reply_count > 0) {
```

Comment:
why remove `thread.is_following` i understand from the PR description `thread.reply_count > 0` removal was needed

Reply:
It's part of what's causing the issue here as well, because when the thread isn't fully loaded `is_following` is set to `false` by default I believe, so it won't try and fully load the thread to figure out whether it's actually being followed or not.

Reply:
ahh i see the defaults

Reply:
@calebroseland It seems as though we are not always getting a value of `is_following` even from the complete thread, we are assuming that the existence of the complete thread object implies that the user is following it. I believe the value is `false` when the thread isn't there, so it won't fetch anything and continue to assume the thread isn't followed.

Maybe there's an API change needed here? Maybe the server needs to be more explicit about whether a thread is actually being followed or not.

Reply:
@devinbinnie I'm not 100% sure where it needs to be invalidated, but I do think the root cause is an HTTP caching issue. 

I can repro this locally if HTTP caching is enabled, but I cannot repro this when caching is disabled (e.g. devtools open).

The I don't think there is a logical issue specifically in ThreadFooter here.

The 404s are a secondary side-effect IMO. If there is a _recently unfollowed_ thread stuck in a cached post list, so that it appears as followed after a quick refresh—ThreadFooter will go fetch it as it thinks it needs to (per the stale `post.is_following`), but it doesn't exist, so `404` 🦖.


This change does relieve the symptom, but I don't think the root cause is addressed. As-is, there will be many `404`s appearing in logs from users just casually browsing channels—that will cause alarm if deployed.

Reply:
@calebroseland Finally got a chance to dig through this some more, and yes there is a caching issue happening here. We don't seem to invalidate the channel post cache when we update the `is_following` flag, so it's not always refreshing correctly.